File: Code/fib.cpp
#include <bits/stdc++.h>
#include <chrono>

using namespace std;

int fibonacci_naive(int n)
{
}

int fibonacci_recursive(int n)
{
}

int fibonacci_dp(int n)
{
}

void solve(int number)
{
    auto start = chrono::high_resolution_clock::now();
    cout << fibonacci_recursive(number) << "\n";
    auto end = chrono::high_resolution_clock::now();

    auto duration = chrono::duration_cast<chrono::microseconds>(end-start);
    cout << "Time elapsed recursive: " << duration.count() << " microseconds" << endl;

    start = chrono::high_resolution_clock::now();
    cout << fibonacci_dp(number) << "\n";
    end = chrono::high_resolution_clock::now();

    duration = chrono::duration_cast<chrono::microseconds>(end-start);
    cout << "Time elapsed DP: " << duration.count() << " microseconds" << endl;

    start = chrono::high_resolution_clock::now();
    cout << fibonacci_naive(number) << "\n";
    end = chrono::high_resolution_clock::now();

    auto duration_ns = chrono::duration_cast<chrono::nanoseconds>(end-start);
    cout << "Time elapsed naive: " << duration_ns.count() << " nanoseconds" << endl;
}

int main(int argc, char** argv)
{
    solve(46);

    return 0;
}

File: Code/fib.py
import time

def fib_recursive(n: int) -> int:
    return n

def fib_dp(n: int) -> int:
    return n

def fib_naive(n: int) -> int:
    return n

def solve(n :int) -> None:

    start = time.time_ns()
    r_rec = fib_recursive(n)
    end = time.time_ns()

    print(f'Fibonacci recursive: {r_rec} Time: {end-start} nanoseconds')

    start = time.time_ns()
    r_dp = fib_dp(n)
    end = time.time_ns()

    print(f'Fibonacci dp: {r_dp} Time: {end-start} nanoseconds')
    
    start = time.time_ns()
    r_naive = fib_naive(n)
    end = time.time_ns()

    print(f'Fibonacci naive: {r_naive} Time: {end-start} nanoseconds')

if __name__ == "__main__":
    solve(46)
